src/components/textarea/__tests__/textarea.test.tsx

```diff
@@ -1,9 +1,19 @@
-import { render } from '@testing-library/react'
+import { render, screen } from '@testing-library/react'
 import { TextArea } from '..'
 
```

Comment:
**note:** I've updated the unit tests to avoid the generic "should match a snapshot". While this kind of unit test provides _some_ value, it's missing out (IMO) on one of the most important aspects of unit tests: _documentation_. As such, I've replaced it with three tests that cover some of the important aspects we would want to document about our TextArea, though one is currently skipped (see code comment).

**aside:** I would not say that these unit tests are hitting the bar of "necessary and sufficient" by themselves, but the addition of the visual snapshots helps get us pretty close as they provide coverage of the invalid, disabled and read-only states, plus the initial sizing of text areas based on the `minRows`, `maxRows` and `rows` props.

I can provide more info on what the terms "necessary and sufficient" mean separately (they are terms we recently introduced to the Console FE team).

Reply:
If you have some docs on this, would be more than happy to add to the wiki